org.idempiere.test/src/org/idempiere/test/model/SalesOrderTest.java

```diff
@@ -1640,4 +1641,73 @@ public void testCreditCheckOrder()
 		assertTrue(info.isError(), info.getSummary());
 		assertEquals(DocAction.STATUS_Invalid, order.getDocStatus());
 	}
+	
+	/**
+	 * Test cases for Prepay Order
+	 */
+	@Test
+	public void testPrepayOrder() {
```

Comment:
this is a test for the use case tried with the ticket, but is also important to add test cases to check that the other use cases are still working fine (backward compatibility)